packages/keymaps/src/browser/keymaps-frontend-contribution.ts

```diff
@@ -86,6 +118,53 @@ export class KeymapsFrontendContribution extends AbstractViewContribution<Keybin
             isVisible: w => this.withWidget(w, () => true),
             execute: w => this.withWidget(w, widget => widget.clearSearch()),
         });
+        commands.registerCommand(KeymapsCommands.COPY_KEYBINDING, {
+            isEnabled: (...args) => this.withItem(() => true, ...args),
+            isVisible: (...args) => this.withItem(() => true, ...args),
+            execute: (...args) => this.withItem(item => this.clipboard.writeText(
+                JSON.stringify(Keybinding.apiObjectify(KeybindingItem.keybinding(item)), undefined, '  ')
+            ), ...args)
+        });
+        commands.registerCommand(KeymapsCommands.COPY_COMMAND_ID, {
+            isEnabled: (...args) => this.withItem(() => true, ...args),
+            isVisible: (...args) => this.withItem(() => true, ...args),
+            execute: (...args) => this.withItem(item => this.clipboard.writeText(item.command.id), ...args)
+        });
+        commands.registerCommand(KeymapsCommands.COPY_COMMAND_TITLE, {
+            isEnabled: (...args) => this.withItem(item => !!item.command.label, ...args),
+            isVisible: (...args) => this.withItem(() => true, ...args),
+            execute: (...args) => this.withItem(item => this.clipboard.writeText(item.command.label!), ...args)
+        });
+        commands.registerCommand(KeymapsCommands.EDIT_KEYBINDING, {
+            isEnabled: (...args) => this.withWidgetItem(() => true, ...args),
+            isVisible: (...args) => this.withWidgetItem(() => true, ...args),
+            execute: (...args) => this.withWidgetItem((item, widget) => widget.editKeybinding(item), ...args)
+        });
+        commands.registerCommand(KeymapsCommands.EDIT_WHEN_EXPRESSION, {
+            isEnabled: (...args) => this.withWidgetItem(item => !!item.keybinding, ...args),
+            isVisible: (...args) => this.withWidgetItem(() => true, ...args),
+            execute: (...args) => this.withWidgetItem((item, widget) => widget.editWhenExpression(item), ...args)
+        });
+        commands.registerCommand(KeymapsCommands.ADD_KEYBINDING, {
+            isEnabled: (...args) => this.withWidgetItem(item => !!item.keybinding, ...args),
+            isVisible: (...args) => this.withWidgetItem(item => !!item.keybinding, ...args),
+            execute: (...args) => this.withWidgetItem((item, widget) => widget.addKeybinding(item), ...args)
```

Comment:
I usually try to order parameters by putting the context (widget) and then something within that context (item) later. Just helps me remember stuff better.